linera-base/src/crypto/mod.rs

```diff
@@ -50,14 +49,19 @@ pub enum CryptoError {
     )]
     IncorrectHashSize(usize),
     #[error(
-        "Byte slice has length {0} but a `PublicKey` requires exactly {expected} bytes",
-        expected = dalek::PUBLIC_KEY_LENGTH,
+        "Byte slice has length {len} but a {scheme} `PublicKey` requires exactly {expected} bytes"
     )]
-    IncorrectPublicKeySize(usize),
+    IncorrectPublicKeySize {
+        scheme: String,
+        len: usize,
+        expected: usize,
+    },
     #[error("Could not parse integer: {0}")]
     ParseIntError(#[from] ParseIntError),
     #[error("secp256k1 error: {0}")]
-    Secp256k1Error(::secp256k1::Error),
+    Secp256k1Error(k256::ecdsa::Error),
+    #[error("could not parse public key: {0}: key point at inifinity.")]
+    Secp256k1PointAtIfinity(String),
```

Comment:
```suggestion
    Secp256k1PointAtInfinity(String),
```

Reply:
Done in [Apply review suggestions.](https://github.com/linera-io/linera-protocol/pull/3407/commits/8f530c26831f602c5a89abc74abc7445a02389d6).